drivers/hmis/app/models/hmis/workflow_execution/engine.rb

```diff
@@ -180,11 +180,20 @@ def evaluate_condition(expression)
       calculator.evaluate!(expression, **defaults.merge(all_submitted_values.transform_keys(&:to_sym)))
     end
 
+    # Evaluate conditions against *all* submitted values. This is important because when we reach a gateway,
```

Comment:
Suggest clarifying that this method returns the values, it doesn't do any evaluation
```suggestion
    # Retrieve merged hash of *all* submitted values across all tasks in the workflow. This is important because when we reach a gateway,
```

Reply:
👍 I think I'd prefer to move the comment above into `evaluate_condition` where it seems more relevant